src/test/java/mate/academy/springbootintro/repository/ShoppingCartRepositoryTest.java

```diff
@@ -0,0 +1,94 @@
+package mate.academy.springbootintro.repository;
+
+import static org.hibernate.validator.internal.util.Contracts.assertNotNull;
+import static org.hibernate.validator.internal.util.Contracts.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import mate.academy.springbootintro.config.CustomMySqlContainer;
+import mate.academy.springbootintro.model.ShoppingCart;
+import mate.academy.springbootintro.repository.shoppingcart.ShoppingCartRepository;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.springframework.test.context.jdbc.Sql;
+import java.util.Optional;
+
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+public class ShoppingCartRepositoryTest {
+
+    private static final Long USER_ID = 3L;
+    private static final String SPRING_DATASOURCE_URL =
+            "spring.datasource.url";
+    private static final String SPRING_DATASOURCE_USERNAME =
+            "spring.datasource.username";
+    private static final String SPRING_DATASOURCE_PASSWORD =
+            "spring.datasource.password";
+    private static final String SPRING_DATASOURCE_DRIVER_CLASS_NAME =
+            "spring.datasource.driver-class-name";
+
+    @Autowired
+    private ShoppingCartRepository shoppingCartRepository;
+
+    @DynamicPropertySource
+    static void configureProperties(DynamicPropertyRegistry registry) {
+        CustomMySqlContainer mysqlContainer = CustomMySqlContainer.getInstance();
+        registry.add(SPRING_DATASOURCE_URL, mysqlContainer::getJdbcUrl);
+        registry.add(SPRING_DATASOURCE_USERNAME, mysqlContainer::getUsername);
+        registry.add(SPRING_DATASOURCE_PASSWORD, mysqlContainer::getPassword);
+        registry.add(SPRING_DATASOURCE_DRIVER_CLASS_NAME, mysqlContainer::getDriverClassName);
+    }

```

Comment:
It should all be in `CustomMySqlContainer` class, with some minor changes. Look at the videos/theoretical part on Mate if you don't know how to do it.